02-syntax.Rmd

```diff
@@ -54,6 +54,34 @@ average <- mean(feet / 12 + inches, na.rm = TRUE)
 average<-mean(feet/12+inches,na.rm=TRUE)
 ```
 
+Put a space after the bang operators (`!!` and `!!!`) but not
+before them. Note that the sequence of exclamation marks must not be interrupted
+with spaces, otherwise the meaning is changed to the logical operator `!`.
```

Comment:
It's not quite that the meaning is changed (at least from R's perspective). It's just confusing.

But it might be better to pull this out into a new chapter? (it'll be very short to start with). I'm worried that it's too esoteric to introduce here.

Reply:
Or do you prefer to drop it completely?

Reply:
I would not mind.

Reply:
I think it's fine to start a new chapter - I'm sure we'll have other tidy eval guidelines in the future.

Reply:
Ok. So if I understand you correctly, we can pull over both new sections I proposed into a new chapter *tidyeval*? Maybe inserting that between the chapter on pipes and the one on code documentation?
Further, should we simplify

> "[...] Note that the sequence of exclamation marks must not be interrupted
> +with spaces, otherwise the meaning is changed to the logical operator `!` [...]"

To

> Don't confuse `!!` and `!!!` with `!`.

?

Reply:
Yes, perfect.